src/telliot_feeds/reporters/rewards/time_based_rewards.py

```diff
@@ -8,19 +9,28 @@
 
 async def get_time_based_rewards(oracle: Tellor360OracleContract) -> int:
     """
-    Time based rewards, as of Tellor360,
-    are the TRB balance in the Oracle contract
-    minus the TRB staking rewards and
-    the totalStakeAmount
+    Reward that will be given if a reporter submits now:
+    TBR is calculated by factoring in the time of the last value to determine when it
+    started accruing and the total rewards balance available in the contract since thats
+    what will be dispersed if the reward is < than whats available in the contract
     """
-
     tbr: int
+    reward: int
+
+    time_of_last_new_value, last_val_status = await oracle.read("timeOfLastNewValue")
+    tbr, tbr_status = await oracle.read("getTotalTimeBasedRewardsBalance")
 
-    tbr, status = await oracle.read("getTotalTimeBasedRewardsBalance")
+    # if any call fails return 0 and a msg that tbr can't calc'd
+    if not tbr_status.ok or not last_val_status.ok:
+        error = tbr_status.error if not tbr_status.ok else last_val_status.error
+        logger.warning(f"Unable to calculate time-based rewards for reporter: {error}")
+        return 0
 
-    if status.ok:
-        return tbr
+    reward = (
+        (TimeStamp.now().ts - time_of_last_new_value) * 5e17
```

Comment:
Why `* 5e17`?
Why `/ 300`?
Can you walk me through the calculation

Reply:
.5 tokens every 5 minutes is the setup in the oracle.  there is a comment on the calculation.

Reply:
Could put `# 0.5 tokens (5e17) dispersed every five min (300 sec)` or something. I'm just saying if I didn't understand it at first pass, others probs won't too.

Reply:
sure, np